fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt

```diff
@@ -55,8 +63,16 @@ class ReviewQualityCheckNetworkMiddleware(
         scope.launch {
             when (action) {
                 FetchProductAnalysis, RetryProductAnalysis -> {
-                    val productReviewState = fetchAnalysis()
+                    val productPageUrl = reviewQualityCheckService.selectedTabUrl()
+                    val productAnalysis = reviewQualityCheckService.fetchProductReview()
+                    val productReviewState = productAnalysis.toProductReviewState()
                     store.updateProductReviewState(productReviewState)
+
+                    store.restoreAnalysingStateIfRequired(
```

Comment:
Just to make sure I'm on the same page here, we don't need an `if` or any conditional here since `reviewQualityCheckService.fetchProductReview()` is a running suspend function, and `store.updateProductReviewState(productReviewState)` won't be called until that finishes? 

So essentially, we're firing off the fetch/re-fetch, and while that's running (since it's non-blocking) we call `restoreAnalysingStateIfRequired` to put the UI back in the reanalysis state (if applicable).

Reply:
> Just to make sure I'm on the same page here, we don't need an if or any conditional here since reviewQualityCheckService.fetchProductReview() is a running suspend function, and store.updateProductReviewState(productReviewState) won't be called until that finishes?

Yes, suspend functions run [sequentially](https://kotlinlang.org/docs/composing-suspending-functions.html#sequential-by-default).

> So essentially, we're firing off the fetch/re-fetch, and while that's running (since it's non-blocking) we call restoreAnalysingStateIfRequired to put the UI back in the reanalysis state (if applicable).
No, when the fetch is completed and the state is updated to `ProductReviewState` from the fetch, then we restore to reanalysis state if required.